### ISSUE TITLE
Some results on apex mixed precision(APM) training and large batch size training for moco-v2

### ISSUE THREAD

**Issue by a1600012888** [open-mmlab/mmselfsup#35]
Hi,  Thanks for providing such an amazing repo on self-supervised learning. 

For SSL, it's important to validate its value on the large-scale dataset, so it would be critical if these algorithms can easily adopt large batch training and mixed-precision training without a huge performance drop. 

My experiment showed APM and large batch did not hurt the MOCO-V2 much, and Apex opt_level=O1 can double the throughput on some types of GPUs. Currently, I'm afraid I couldn't speed to much time integrating apex into this repo, I am only providing some basic results on APM & large batch training on MOCO V2.

|  ImageNet1k | AMP \& Large batch  | Baseline  |   
|---|---|---|
| Batch size  | 4096  |  256 |    
| Mixed Precision Training  |  **Apex opt_level: O1** |  FP32  |    
| Init Lr  |  0.48 |  0.03 |   
| WarmUp  |  No |  No |    
| Pre-training Loss  |  6.624 |  6.572 |   
| ImageNet Linear protocol  |  **66.60** |  67.28 |

**Comment by XiaohangZhan** [comment]
It's so great to hear this news. Besides, have you tried warmup and LARS? Typically they shall make improvements. And what's the speed gain of apex? The speed bottleneck of mocov2 may lie in data processing on CPU, since mocov2 adopts gaussian blur with a large kernel.

**Comment by a1600012888** [comment]
> It's so great to hear this news. Besides, have you tried warmup and LARS? Typically they shall make improvements. And what's the speed gain of apex? The speed bottleneck of mocov2 may lie in data processing on CPU, since mocov2 adopts gaussian blur with a large kernel.

Right, warmup, more smooth learing rate and lars are common methods for large batch training.
As for the speed gain, we can nearly double the speed for MocoV2 with ResNet152 backbone, but for the smaller ResNet50 backbone, we could only boost the speed by 40%. (On T4. https://www.techpowerup.com/gpu-specs/tesla-t4.c3316)

**Comment by XiaohangZhan** [comment]
Great! Would you mind contributing to integrating AMP into openselfsup? You may raise a pull request, in this way, your contribution will be recorded.
